app/battle/fight.py

```diff
@@ -0,0 +1,21 @@
+from app.knights.knight import Knight
+
+def calculate_damage(attacker: Knight, defender: Knight) -> int:
+    return max(
+        0,
+        attacker.stats["power"] - defender.stats["protection"]
+    )
+
+def fight(knight1: Knight, knight2: Knight) -> dict:
+    # Knight 1 attacks Knight 2
+    damage_to_knight2 = calculate_damage(knight1, knight2)
+    knight2.receive_damage(damage_to_knight2)
+
+    # Knight 2 attacks Knight 1
+    damage_to_knight1 = calculate_damage(knight2, knight1)
+    knight1.receive_damage(damage_to_knight1)

```

Comment:
Similar to the previous comment, ensure that the `Knight` class has a `receive_damage` method. If it doesn't exist, this line will cause an error. Verify the method's implementation to ensure it correctly updates the knight's health points.